lib/avro/builder/types/record_type.rb

```diff
@@ -17,6 +20,10 @@ def initialize(name = nil, options: {}, cache:, field: nil, &block)
           instance_eval(&block) if block_given?
         end
 
+        def dsl_method?(name)
+          DSL_METHODS.include?(name)
```

Comment:
It's only called via `method_missing`/`respond_to_missing?` which call it with a symbol, so I think this is fine as is.